### ISSUE TITLE
ScaFi currentPosition method throws java.lang.ClassCastException 

### ISSUE THREAD

**Issue by cric96** [AlchemistSimulator/Alchemist#723]
In ScaFi incarnation, currentPosition should return a Point3D object. Indeed, in **ScafiIncarnationForAlchemist** the  P type is overridden with:
```  override type P = Point3D```

But, the sensor for the position is set with AlchemistPosition type (**RunScafiProgram**):
```
sealed class RunScafiProgram[T, P <: Position[P]] (....) extends AbstractLocalAction[T](node) {
  override def execute() {
  ....
    val position: P = environment.getPosition(node)
  ....
}
```
Is it right to map Alchemist position in ScaFi point? In that case, can I change P with Point3D? @metaphori there were some reasons for this choice?

**Comment by metaphori** [comment]
Should be fixed in https://github.com/AlchemistSimulator/Alchemist/pull/711
I chose to map Alchemist stuff to ScaFi when possible, but also keep Alchemist-related sensors

**Comment by metaphori** [comment]
The quickest fix would be to override the method and correct the way the sensor is accessed

**Comment by DanySK** [comment]
#711 is already released in 10.0.1.

**Comment by DanySK** [comment]
@cric96 close the issue if it is solved, document otherwise
